near-sdk/src/types/event.rs

```diff
@@ -0,0 +1,95 @@
+pub trait StandardEvent {
+    fn format(&self) -> String;
+    fn emit(&self);
+}
+
+pub struct Event {}
+impl Event {
+    pub fn emit(standard_event: impl StandardEvent) {
+        crate::env::log_str(&format!("EVENT_JSON:{}", &standard_event.format()));
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+
+    use crate::test_utils::get_logs;
+    use crate::{near_bindgen, AccountId};
+
+    use super::Event;
+    use super::StandardEvent;
+    use crate as near_sdk;
+
+    #[near_bindgen(events)]
+    pub enum TestEvents<'a, 'b, T>
+    where
+        T: crate::serde::Serialize + Clone,
+    {
+        #[event_meta(standard = "swap_standard", version = "1.0.0")]
```

Comment:
Curious why the decision to group all events in one enum? Would this not just be simpler if it was an attribute on a struct? I suppose this is a bit opinionated, but wondering if there is a reason this _must_ be an enum

Reply:
nope, that's fine. It's opinionated, was just sharing my intuition. I suppose we could also just support _both_ if it doesn't increase complexity meaningfully.

cc @itegulov wdyt?

Reply:
I treat events as some sort of standard and an enum in my head represents the fact that there is a finite number of types of events that a contract can emit. That said, I would consider extracting `standard = "..."` to the enum declaration level instead of the enum variants level. This way we could have FtEvents, NftEvents, MtEvents, FayyrEvents, etc, so one could import and use them in a composable way.

Reply:
> @itegulov wdyt about this? Do you think the implications of enum vs struct annotation would have any implication on integrating into the ABI?

I don't think it matters, we should be able to generate ABI for both options.

Personally, I find that having a mega-enum covering all standards used in this contract is a bit of an antipattern, but not a strong opinion. What @frol suggested makes more sense to me - having an enum per standard looks cleaner.